### PR TITLE
Fix for issue #45

### PR DESCRIPTION
**Pull request by noofherder**
Several bugs (I think) found in BKDNode and fixed. I think.

### CODE REVIEW

File: src/main/java/com/github/flaxsearch/api/BKDNode.java [hunk 1]
```diff
@@ -63,24 +65,31 @@ public void setParent(BKDNode parent) {
...
 
-    public BKDNode findParent(BKDNode node, int numDims, int bytesPerDim) {
-        // go up the tree until we find a node whose parent
-        // contains this node
+    public BKDNode findParentOf(BKDNode node, int numDims, int bytesPerDim) {
+        // go up the tree until we find a node which contains the supplied node
         BKDNode parent = this;
-        while (parent.contains(node, numDims, bytesPerDim) == false) {
+        while (! parent.contains(node, numDims, bytesPerDim)) {
             parent = parent.parent;
```
Comment (romseygeek) [inline src/main/java/com/github/flaxsearch/api/BKDNode.java]: Can we not change these?  I know IntelliJ does this, but I think it makes things a lot less clear (! is easy to miss, '== false' isn't)

**Review by romseygeek — COMMENTED** [review]
Looks good, but Travis is unhappy about something

**Comment by noofherder** [comment]
It wasn't IntelliJ, it was me and my odd preferences. :P  I'll change it back.

File: src/main/java/com/github/flaxsearch/api/BKDNode.java [hunk 2]
```diff
@@ -74,13 +75,21 @@ public BKDNode findParent(BKDNode node, int numDims, int bytesPerDim) {
...
         for (int i = 0; i < numDims; i++) {
             int offset = i * bytesPerDim;
-            contains &= (StringHelper.compare(bytesPerDim, minPackedValue, offset, node.minPackedValue, offset) > 0 &&
-                    StringHelper.compare(bytesPerDim, maxPackedValue, offset, node.maxPackedValue, offset) < 0);
+            int cMin = StringHelper.compare(bytesPerDim, minPackedValue, offset,
+                                            node.minPackedValue, offset);
+            int cMax = StringHelper.compare(bytesPerDim, maxPackedValue, offset,
+                                            node.maxPackedValue, offset);
+            if (cMin > 0 || cMax < 0) return false;
         }
```
Comment (romseygeek) [inline src/main/java/com/github/flaxsearch/api/BKDNode.java:85]: Can you add a test for this?  I should have put one in there in the first place, really...

**Pull request merged**